cypress/utils/dashboards/workspace-plugin/commands.js

```diff
@@ -0,0 +1,128 @@
+    url: `${BASE_PATH}${WORKSPACE_API_PREFIX}/_list`,
+    headers: {
+      'osd-xsrf': true,
+    },
+    body: {},
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      resp.body.result.workspaces.map(({ name, id }) => {
+        if (workspaceName === name) {
+          cy.request({
+            method: 'DELETE',
+            url: `${BASE_PATH}${WORKSPACE_API_PREFIX}/${id}`,
+            headers: {
+              'osd-xsrf': true,
+            },
+          });
+        }
+      });
+    }
+  });
+});
+
+Cypress.Commands.add('createWorkspace', (workspace) => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_PATH}${WORKSPACE_API_PREFIX}`,
+    headers: {
+      'osd-xsrf': true,
+    },
+    body: {
+      attributes: {
+        name: workspace.name,
+        description: 'test_description',
+      },
+    },
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      workspace.id = resp.body.result.id;
+    } else {
+      throw new Error(`Create workspace ${workspace} failed!`);
+    }
+  });
+});
+
+Cypress.Commands.add('checkWorkspace', (workspaceName, expected) => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_PATH}${WORKSPACE_API_PREFIX}/_list`,
+    headers: {
+      'osd-xsrf': true,
+    },
+    body: {},
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      let found = false;
+      resp.body.result.workspaces.map(
+        ({ name, description, features, permissions }) => {
+          if (workspaceName === name) {
+            if (description !== expected.description) {
+              throw new Error(
+                `workspace ${workspaceName} is not as expected, expected description is ${expected.description}, but is ${description}`
+              );
+            } else {
+              expected.features.forEach((feature) => {
+                if (!features.includes(feature)) {
+                  const expectedFeatures = JSON.stringify(expected.features);
+                  const actualFeatures = JSON.stringify(features);
+                  throw new Error(
+                    `workspace ${workspaceName} is not as expected, expected features are: ${expectedFeatures}, but are: ${actualFeatures}`
+                  );
+                }
+              });
+
+              if (permissions && expected.permissions) {
+                const expectedPermissions = JSON.stringify(
+                  expected.permissions
+                );
+                const actualPermissions = JSON.stringify(permissions);
+                Object.entries(permissions).forEach(([key, value]) => {
+                  if (!expected.permissions[key]) {
+                    throw new Error(
+                      `permissions for workspace ${workspaceName} is not as expected, expected are:  ${expectedPermissions}, but are ${actualPermissions}`
+                    );
+                  } else {
+                    if (expected.permissions[key].users) {
```

Comment:
Shall we separate a string array compare function, then we can reused in users array and groups array. This function will check array length and all elements are equal.

Reply:
I've added that functions, please help to take a look, thank you!